types/coin.go

```diff
@@ -392,10 +388,7 @@ func (coins Coins) negative() Coins {
 	res := make([]Coin, 0, len(coins))
 
 	for _, coin := range coins {
-		res = append(res, Coin{
-			Denom:  coin.Denom,
-			Amount: coin.Amount.Neg(),
-		})
+		res = append(res, NewCoin(coin.Denom, coin.Amount.Neg()))
 	}
```

Comment:
This method is used to create negative coins on purpose (don't know why we want to use something like that).  So calling NewCoin() here will panic.